internal/services/engines/nodejs/rules.go

```diff
@@ -50,7 +50,7 @@ func NewNoUseEval() text.TextRule {
 		},
 		Type: text.Regular,
 		Expressions: []*regexp.Regexp{
-			regexp.MustCompile(`(eval\(.+)(?:req\.|req\.query|req\.body|req\.param)`),
+			regexp.MustCompile(`\W(eval\()`),
```

Comment:
This regex will not match if the `eval` does not contains any spaces before. 

Also I think that would be better that this rule only match use of eval that is not a string, anything else should report a vulnerability.

We can change the regex to`(eval\((\'|").*('|")\))` and change the type of rule to `text.NotMatch`.

Using this any use of eval that is not **only** a raw string will report a vulnerability, for example:
- `eval(input)`
- `eval("bash -c" + input)`

What do you think? 

cc @nathanmartinszup @wiliansilvazup @iancardosozup

Reply:
I changed it to this format, precisely because they have several possibilities, for example:

```
.eval(`${bar}-foo`)
.eval('foo' + bar)
.eval("foo" + bar)
```

How would these scenarios look like?

Therefore, I made accusse if any occurrence of `eval(` decreases confidence.

Reply:
The intention is precisely to accuse only in case there is a space or special character before the `eval` to avoid false positives like:

`fooeval()`

Reply:
I think I got confused and `text.NotMatch` doesn't help in this case. The idea would be that any use of eval that **not** is a string would point to vulnerability (so it should take those cases you showed above) but `NotMatch` doesn't work that way.

I think at first then we can leave any use of eval as a possible vulnerability anyway.

I think I would have to remove the `\W` from the regex to match if the eval is at the beginning of the line (without spaces).

Reply:
Yes, I am using this site.
Yes, we can remove it, I was just afraid to see some function name that ends with `eval`, but I believe it is unlikely.

Reply:
Hey @nathanmartinszup  this regex no consider this example:

`eval(foo)`